test/e2e/page-objects/benchmark/page-load-benchmark.ts

```diff
@@ -0,0 +1,471 @@
+  ): Promise<BenchmarkResult> {
+    const page = await this.context?.newPage();
+
+    if (!page) {
+      throw new Error('Browser Context not initialized.');
+    }
+
+    // Enable performance monitoring
+    await page.addInitScript(() => {
+      const originalNow = performance.now;
+      const startTime = Date.now();
+      performance.now = () => originalNow() + (Date.now() - startTime);
+    });
+
+    await page.goto(url, { waitUntil: 'networkidle' });
+
+    // Wait for page to be fully loaded
+    await page.waitForLoadState('domcontentloaded');
+    await page.waitForLoadState('networkidle');
+
+    // Collect performance metrics
+    const metrics: BenchmarkMetrics = await page.evaluate(() => {
+      const navigation = performance.getEntriesByType(
+        'navigation',
+      )[0] as PerformanceNavigationTiming;
+      const paint = performance.getEntriesByType('paint');
+      const lcp = performance.getEntriesByType(
+        'largest-contentful-paint',
+      )[0] as PerformanceEntry;
+
+      return {
+        pageLoadTime: navigation.loadEventEnd - navigation.startTime,
+        domContentLoaded:
+          navigation.domContentLoadedEventEnd - navigation.startTime,
+        firstPaint: paint.find((p) => p.name === 'first-paint')?.startTime || 0,
+        firstContentfulPaint:
+          paint.find((p) => p.name === 'first-contentful-paint')?.startTime ||
+          0,
+        largestContentfulPaint: lcp?.startTime || 0,
+        memoryUsage: performance.memory
+          ? {
+              usedJSHeapSize: performance.memory.usedJSHeapSize,
+              totalJSHeapSize: performance.memory.totalJSHeapSize,
+              jsHeapSizeLimit: performance.memory.jsHeapSizeLimit,
+            }
+          : undefined,
+      };
+    });
+
+    const result: BenchmarkResult = {
+      page: url,
+      run: runNumber,
+      metrics,
+      timestamp: new Date().getTime(),
+    };
+
+    this.results.push(result);
+    await page.close();
+
+    return result;
+  }
+
+  /**
+   * Executes the complete benchmark test suite across multiple browser loads and page loads.
+   * Creates fresh browser contexts for each browser load to simulate real-world conditions,
+   * and measures each URL multiple times to gather statistical data.
+   *
+   * @param urls - Array of URLs to test
+   * @param browserLoads - Number of fresh browser contexts to create (default: 10)
+   * @param pageLoads - Number of page loads per browser context (default: 10)
+   */
+  async runBenchmark(
+    urls: string[],
+    browserLoads: number = 10,
+    pageLoads: number = 10,
+  ) {
+    console.log(
+      `Starting benchmark: ${browserLoads} browser loads, ${pageLoads} page loads per browser`,
+    );
+
+    for (let browserLoad = 0; browserLoad < browserLoads; browserLoad++) {
+      console.log(`Browser load ${browserLoad + 1}/${browserLoads}`);
+
+      // Setup fresh browser context for each browser load
+      if (browserLoad > 0) {
```

Comment:
thoughts on always calling `await this.context?.close();` at the end of the loop and always calling `this.context = ..., etc` at the start of the loop rather than using this guard?